spark/src/main/kotlin/com/adevinta/spark/components/textfields/DefaultSparkTextFieldColors.kt

```diff
@@ -0,0 +1,375 @@
+        }
+        return if (enabled) {
+            animateColorAsState(targetValue, tween(durationMillis = 150))
+        } else {
+            rememberUpdatedState(targetValue)
+        }
+    }
+
+    /**
+     * Represents the container color for this text field.
+     */
+    @Composable
+    internal fun containerColor(): State<Color> {
+        return rememberUpdatedState(containerColor)
+    }
+
+    /**
+     * Represents the color used for the placeholder of this text field.
+     *
+     * @param enabled whether the text field is enabled
+     */
+    @Composable
+    internal fun placeholderColor(enabled: Boolean): State<Color> {
+        return rememberUpdatedState(if (enabled) placeholderColor else disabledPlaceholderColor)
+    }
+
+    /**
+     * Represents the color used for the label of this text field.
+     *
+     * @param enabled whether the text field is enabled
+     * @param isError whether the text field's current value is in error
+     * @param interactionSource the [InteractionSource] of this text field. Helps to determine if
+     * the text field is in focus or not
+     */
+    @Composable
+    internal fun labelColor(
+        enabled: Boolean,
+        isError: Boolean,
+        interactionSource: InteractionSource,
+    ): State<Color> {
+        val focused by interactionSource.collectIsFocusedAsState()
+
+        val targetValue = when {
+            !enabled -> disabledLabelColor
+            isError -> errorLabelColor
+            focused -> focusedLabelColor
+            else -> unfocusedLabelColor
+        }
+        return rememberUpdatedState(targetValue)
+    }
+
+    @Composable
+    internal fun textColor(enabled: Boolean): State<Color> {
+        return rememberUpdatedState(if (enabled) textColor else disabledTextColor)
+    }
+
+    @Composable
+    internal fun supportingTextColor(
+        enabled: Boolean,
+        isError: Boolean,
+        interactionSource: InteractionSource,
+    ): State<Color> {
+        val focused by interactionSource.collectIsFocusedAsState()
+
+        return rememberUpdatedState(
+            when {
+                !enabled -> disabledSupportingTextColor
+                isError -> errorSupportingTextColor
+                focused -> focusedSupportingTextColor
+                else -> unfocusedSupportingTextColor
+            },
+        )
+    }
+
+    /**
+     * Represents the color used for the cursor of this text field.
+     *
+     * @param isError whether the text field's current value is in error
+     */
+    @Composable
+    internal fun cursorColor(isError: Boolean): State<Color> {
+        return rememberUpdatedState(if (isError) errorCursorColor else cursorColor)
+    }
+
+    /**
+     * Represents the colors used for text selection in this text field.
+     */
+    internal val selectionColors: TextSelectionColors
+        @Composable get() = textSelectionColors
+
+    override fun equals(other: Any?): Boolean {
```

Comment:
Any particular reason to not take advantage of Kotlin data classes here?

Reply:
I can't answer you for this as it's a copy of the OutlineTextfield from Material to support helper text and char counter :-/